File: uberpolylineanimation/src/main/java/com/logicbeanzs/uberpolylineanimation/RouteEvaluator.java
package com.logicbeanzs.uberpolylineanimation;

import android.animation.TypeEvaluator;

import com.google.android.gms.maps.model.LatLng;

/**
 * Created by pritam on 22/12/16.
 */

public class RouteEvaluator implements TypeEvaluator<LatLng> {
    @Override
    public LatLng evaluate(float t, LatLng startPoint, LatLng endPoint) {
        double lat = startPoint.latitude + t * (endPoint.latitude - startPoint.latitude);
        double lng = startPoint.longitude + t * (endPoint.longitude - startPoint.longitude);
        return new LatLng(lat,lng);
    }
}

File: uberpolylineanimation/src/main/java/com/logicbeanzs/uberpolylineanimation/RouteEvaluator.kt
package com.logicbeanzs.uberpolylineanimation

import android.animation.TypeEvaluator
import com.google.android.gms.maps.model.LatLng

/**
 * Created by pritam on 22/12/16.
 */
class RouteEvaluator : TypeEvaluator<LatLng> {
    override fun evaluate(t: Float, startPoint: LatLng, endPoint: LatLng): LatLng {
        val lat = startPoint.latitude + t * (endPoint.latitude - startPoint.latitude)
        val lng = startPoint.longitude + t * (endPoint.longitude - startPoint.longitude)
        return LatLng(lat, lng)
    }
}